simulator/infra/memory/memory.h

```diff
@@ -49,7 +49,7 @@ class FuncMemory
         template<typename T>
         void write( T value, Addr addr, T mask = all_ones<T>());
 
-        size_t memcpy_host_to_guest( Addr dst, const Byte* src, size_t size);
+        virtual size_t memcpy_host_to_guest( Addr dst, const Byte* src, size_t size);
```

Comment:
This method does not have to be virtual.

Reply:
clang-tidy warns about overriding non-virtual method

Reply:
Yes, because `final` and `override` keywords are available only for virtual functions. There is no sense to use them with non-virtual functions.

Reply:
At first, clang-tidy warns about overriding a non-virtual method (without `final`); then it asks to put `final` or `override`.

Reply:
Let's handle that later. Please provide the Clang-Tidy output, it is something wrong.

Reply:
```powershell
PS D:\Dev\test> cat .\test.cpp
struct A {
    int func() { return 0; }
};

struct B : A {
    int func() { return 1; }
};

PS D:\Dev\test> clang-tidy.exe --version
LLVM (http://llvm.org/):
  LLVM version 6.0.0
  Optimized build.
  Default target: x86_64-pc-windows-msvc
  Host CPU: westmere
PS D:\Dev\test> clang-tidy.exe test.cpp -checks='*' -- -std=c++11
PS D:\Dev\test> clang-tidy.exe test.cpp -checks='*' -- -std=c++17
PS D:\Dev\test>
```

Reply:
I mixed up a bit: these warnings are not by clang-tidy, they are by my IDE.